### DISCUSSION TITLE
Persistence of live images SpiralLinux

### DISCUSSION THREAD

**Discussion by Raider727** [SpiralLinux/SpiralLinux-project#51] [General]
GeckoLinux has Persistence  of the live images . SpiralLinux dose not seem to Can I get SpiralLinux live iso to be  Persistence ?

**Comment by geckolinux** [comment]
Hi there, thanks for the question. The topic of persistence came up here: https://github.com/SpiralLinux/SpiralLinux-project/discussions/8

**Comment by ghost** [comment]
The original topic was locked but I think I found an issue and maybe this could help Raider727 and others
Regarding the linked topic: https://github.com/orgs/SpiralLinux/discussions/8 .
As a newbie, I think there might be a typo/possible variant or difference:


Step 2 in the instruction ping-wu gave on the linked topic should probably be:


sudo mount ***0628_x86-64.iso /mnt
rsync -r --progress /mnt/ /run/media/$USER/EFI/


P.S.
You may also chose to use debian's persistence instructions for ventoy if so desired, I still wanted to try accomplishing it the ay ping-wu said, too.
Also, for this way you should have rsync installed - I think it's not installed by default on some linux OS'es. 
After installing rsync as I didn't have it and fixing the noted typo allegedly, this now seems to advance in live system creation.

Also I may have a size issue for the EFI partiation - will try to solve that too, probably there are errors on that USB or 4GB is a little too small otherwise or it's related to not having a USB 3.0+ to test with yet.

     87,916,544  88%    4.30MB/s    0:00:02  rsync: [receiver] write failed on "/run/media/<myusername>/EFI/live/initrd.img-5.18.0-0.deb11.4-amd64": No space left on device (28)
rsync error: error in file IO (code 11) at receiver.c(379) [receiver=3.2.5]

A late edit: **removed a part of line from this post which had caused this error** (This is explaind bellow - no information was lost) to not mislead/cause issues.

**Reply by ghost** [reply]
I found an issue with my post above, it was very newbie-like from my side: It looks like I have both the image and the contents inside EFI partition now. Will try to redo the process without having the image itself there and will be back to report. Maybe that's why I run out of place.

I edited that part out to avoid people encountering the same error as I did, above. However the info regarding the cause of this error is found in this comment.

**Reply by ghost** [reply]
No error this time on step 2. Since the rest is taking me time, updating now.

Noting to help other newcommers: 
While following ping-wu guide, do not copy the iso into your EFI partition (even solely for this step) else you might run out of space.
Also, you can mount the iso from any point but the rsync command would better run from within the EFI partition.

**Reply by ghost** [reply]
Update: 

In the instructions on the topic https://github.com/orgs/SpiralLinux/discussions/8 by ping-wu
On step **3, 3a and 3b** one also might have to add /run just before the /media
This should be at least done if adding /run before /media solved the issue in **step 2**.

I am sorry for my many posts. (I somehow couldn't edit some initially.)

Last Update: It worked! Network connectivity details were saved so I know persistence also works, not only booting.
Thank you ping-wu for guiding us in making a live persistence SpiralLinux USB, and thank you GeckoLinux for this nice OS I now try.

**Reply by geckolinux** [reply]
Thanks a lot @Visiting-Newbie , I appreciate it.
